index.js

```diff
@@ -1,262 +1,515 @@
-    },
-    /**
-     * ## `null`
-     * No authentication.
-     #
-     * ## GSSAPI
-     * Kerberos
-     *
-     * ## PLAIN
-     * LDAP
-     *
-     * ## DEFAULT
-     * driver will autoselect SCRAM-SHA-1 | MONGODB-CR
-     * depending on the servers capabilities.
-     * @see https://github.com/mongodb/node-mongodb-native/pull/1299
-     */
-    auth_mechanism: {
-      type: 'string',
-      values: [
-        null,
-        'DEFAULT', 'SCRAM-SHA-1', 'MONGODB-CR',
-        'MONGODB-X509',
-        'GSSAPI',
-        'PLAIN'
-      ],
-      default: null
-    },
-    gssapi_service_name: {
-      type: 'string',
-      default: null
-    },
-    /**
-     * a.k.a `authdb`
-     */
-    database_name: {
-      type: 'string',
-      default: 'admin'
-    },
-    ssl: {
-      type: 'boolean',
-      default: false
-    },
-    /**
-     * Validate mongod server certificate against ca
-     * (needs to have a mongod server with ssl support, 2.4 or higher).
-     */
-    ssl_validate: {
-      type: 'boolean',
-      default: false
-    },
-
-    /**
-     * Array of valid certificates either as Buffers or Strings
-     * (needs to have a mongod server with ssl support, 2.4 or higher).
-     */
-    ssl_ca: {
-      type: 'array',
-      default: null
-    },
-
-    /**
-     * String or buffer containing the certificate we wish to present
-     * (needs to have a mongod server with ssl support, 2.4 or higher).
-     */
-    ssl_cert: {
-      type: 'string',
-      default: null
-    },
-
-    /**
-     * String or buffer containing the certificate private key we wish to present
-     * (needs to have a mongod server with ssl support, 2.4 or higher).
-     */
-    ssl_private_key: {
-      type: 'string',
-      default: null
-    },
-
-    /**
-     * String or buffer containing the certificate password
-     * (needs to have a mongod server with ssl support, 2.4 or higher).
-     */
-    ssl_private_key_password: {
-      type: 'string',
-      default: null
+ * # Top-Level
+ */
+assign(props, {
+  /**
+   * User specified name for this connection.
+   *
+   * @example
+   *   My Laptop
+   *   PRODUCTION
+   *   Analyics Box
+   */
+  name: {
+    type: 'string',
+    default: 'Local'
+  },
+  hostname: {
+    type: 'string',
+    default: 'localhost'
+  },
+  port: {
+    type: 'number',
+    default: 27017
+  }
+});
+
+assign(derived, {
+  /**
+   * @see http://npm.im/mongodb-instance-model
+   */
+  instance_id: {
+    deps: ['hostname', 'port'],
+    fn: function() {
+      return format('%s:%s', this.hostname, this.port);
     }
+  }
+});
+
+/**
+ * ## Authentication
+ */
+assign(props, {
+  /**
+   * `auth_mechanism` for humans.
+   */
+  authentication: {
+    type: 'string',
+    values: [
+      /**
+       * Use no authentication.
+       */
+      'NONE',
+      /**
+       * Allow the driver to autodetect and select SCRAM-SHA-1
+       * or MONGODB-CR depending on server capabilities.
+       */
+      'MONGODB',
```

Comment:
Why is this called "MONGODB?" Would something like CHALLENGERESPONSE or USERNAMEPASSWORD be a better name?